### DISCUSSION TITLE
How to set Gtk Window size in Uno 5?

### DISCUSSION THREAD

**Discussion by HakanL** [unoplatform/uno#13806] [Q&A] [answered]
In Uno 4.10 I used to set the Gtk window size to not resizable, and a fixed size (to mimic my hardware) by setting GtkHost.Window.Resizable and Size, because GtkHost.Window was a static. That was changed in 5.0, so my question is how can I do this now? I tried the ApplicationView, but it doesn't do anything. What's the preferred way to set the Gtk Window size in Uno5? I've only tested this on Gtk Skia on Windows.

**Comment by MartinZikmund** [comment]
You should be able to utilize an approach similar to https://stackoverflow.com/a/16703307 . However, I will add a way to access the native window instance via the Uno `Window` instance, created an issue for this

**Reply by HakanL** [reply]
Thanks, I'll just wait for you to add an instance, I'm not planning to fully use Uno5 yet. But wouldn't it be a good idea to use the ApplicationView?

**Comment by TypoMustakes** [comment]
Wow, really old thread. Is there an update on this? Is there a way to access the window instance now?

**Comment by MartinZikmund** [comment, marked as answer]
Native window instance is now accessible via https://platform.uno/docs/articles/features/windows-ui-xaml-window.html#retrieving-the-native-window

That being said, GTK is now in legacy support mode, and it is preferrable to utilize the new desktop targets instead.

CC @MiskolcziRichard

**Reply by lindexi** [reply]
Thank you @MartinZikmund . But I find it can not move or resize the x11 window by `AppWindow.MoveAndResize`.

**Reply by MartinZikmund** [reply]
My fault, need to finish implementation of this, related PR is in progress, will finalize ASAP
